### PR TITLE
ARKit Planes, World Alignment, reload configuration fix

### PR DESCRIPTION
**Pull request by pierotofy**
ARKit Planes, World Alignment, reload configuration fix

This PR adds the following:

* ARKit planes detection
* ARKit world alignment setting
* Fixes a bug during ARKit session restart (via _reload) by changing self.arConfig instead of creating a new ARWorldTrackingConfiguration object. The current behavior makes any call to ExpoTHREE.setIsLightEstimationEnabled and ExpoTHREE.setIsPlaneDetectionEnabled void (nothing happens, for example, if you attempt to disable light estimation by calling ExpoTHREE.setIsLightEstimationEnabled(arSession, false), successive calls to ExpoTHREE.getARLightEstimation(arSession) happily continue to return estimation values.

Being my first PR, please let me know if I need to change anything.

Thanks, feedback is much appreciated.

### CODE REVIEW

**Comment by pierotofy** [comment]
Related PR in ExpoTHREE: https://github.com/expo/expo-three/pull/13

File: ios/Exponent/Versioned/Modules/Api/GL/ARKit/EXGLARSessionManager.m [hunk 1]
```diff
@@ -248,6 +251,42 @@ - (NSDictionary *)rawFeaturePoints
...
+- (NSDictionary *)planes
+{
+  if (!self.arSession) {
+    return nil;
+  }
+  
+  NSArray<ARAnchor *> *anchors = self.arSession.currentFrame.anchors;
+  NSMutableArray *planes = [NSMutableArray array];
+  
+  for (int i = 0; i < anchors.count; i++) {
```
Comment (terribleben) [inline ios/Exponent/Versioned/Modules/Api/GL/ARKit/EXGLARSessionManager.m]: Why not `for ... in` ?

**Review by terribleben — COMMENTED** [review]
lgtm, will wait for bacon to weigh in!

File: ios/Exponent/Versioned/Modules/Api/GL/ARKit/EXGLARSessionManager.m (hunk 1 above)
Comment (pierotofy) [reply ios/Exponent/Versioned/Modules/Api/GL/ARKit/EXGLARSessionManager.m]: Forgot about fast enums, yes it would be better that way.

**Pull request merged**